library/core/src/escape.rs

```diff
@@ -149,77 +158,178 @@ const fn escape_unicode<const N: usize>(c: char) -> ([ascii::Char; N], Range<u8>
     (output, (start as u8)..(N as u8))
 }
 
-/// An iterator over an fixed-size array.
-///
-/// This is essentially equivalent to array’s IntoIter except that indexes are
-/// limited to u8 to reduce size of the structure.
-#[derive(Clone, Debug)]
-pub(crate) struct EscapeIterInner<const N: usize> {
-    // The element type ensures this is always ASCII, and thus also valid UTF-8.
-    data: [ascii::Char; N],
-
-    // Invariant: `alive.start <= alive.end <= N`
+#[derive(Clone, Copy)]
+union MaybeEscapedCharacter<const N: usize> {
+    pub escape_seq: [ascii::Char; N],
+    pub literal: char,
```

Comment:
What is the value of this being a union? Assuming `N >= 4`, it seems like any `char`s could just be written to `escape_seq` to eliminate the union unsafety.

Reply:
`ascii::Char` only ranges from 0 to 127. So you would have to store both in `[u8; N]` instead, and transmute both variants.